src/utils/error-utils.ts

```diff
@@ -0,0 +1,9 @@
+import logger from "@root/logger/logger"
+
+export default function createErrorAndLog<T extends Error>(
+  ErrorType: new (errorMessage: string) => T,
+  message: string
+): T {
+  logger.error(message)
+  return new ErrorType(message)
+}
```

Comment:
sadly this doesn't lend itself well to extension because errors are very likely to take additional arguments. 

if we want to always ensure errors are logged at initialisation, we could always do `logger.error(message)` but this might be verbose and overkill, especially when errors are expected.

Reply:
hmm true, but that could be solved with 1c4bb481442be753945a061fc33687acff938dd2 right? 

> if we want to always ensure errors are logged at initialisation, we could always do logger.error(message) but this might be verbose and overkill, especially when errors are expected.

nah ya I want this to be flexible to dev, I dont think logging every error should be the way to go here.

Reply:
sadly, i don't actually think that works ._. it's important to think about whether the increased verbosity is worth the trade off (here, the verbosity allows us flexibility). 

i think in this case, the LOC saved is probably like 1-2 lines, which isn't much. i think if we do want the method, and i can see why, we might wish to scope it to only a very specific file or module because we can't enforce the error shape that the util method wants. 

to show the benefits of the util method, consider the 2 cases below - 
```ts
const err = new err(createErrorAndLog(...)) // 1 line

const error = new SomeExampleError(...)
logger.error(err.message)
return err(error) // 3 lines
```

Reply:
removing this bit here to unblock this PR, think this warrents a separate discussion in another thread! 3f2ead83ed79d6df958523103ce0767f1e6f35df